Project/app/src/main/java/org/ionproject/android/loading/LoadingActivity.kt

```diff
@@ -24,16 +35,61 @@ class LoadingActivity : ExceptionHandlingActivity() {
         setContentView(R.layout.activity_loading)
         linearlayout_activity_loading.addGradientBackground()
 
-        // If JsonHome contains all required resources then open main activity else app must be outdated
+        /**
+         * If JsonHome contains all required resources then open main activity else app must be outdated
+         *
+         * We check if remote config live data is null bc this observer is used for all requests made to
+         * the API: if it is the first request ever made, get the remote config
+         * (the live data is null at this point); if it is the request after getting remote config,
+         * and it fails again, redirect to ISEL's page
+         *
+         */
         loadingViewModel.observeRootLiveData(this) {
-            if (it != null) {
-                val intent = Intent(this, MainActivity::class.java)
-                intent.putExtra(ROOT_KEY, it)
-                this.startActivity(intent)
-            } else {
-                val intent = Intent(this, ErrorActivity::class.java)
-                this.startActivity(intent)
+            when (it) {
+                is FetchSuccess<Root> -> {
+                    val intent = Intent(this, MainActivity::class.java)
+                    intent.putExtra(ROOT_KEY, it.value)
+                    this.startActivity(intent)
+                }
+                is FetchFailure<Root> -> {
+                    if (loadingViewModel.getRemoteConfigLiveData() == null) {
+                        loadingViewModel.getRemoteConfig()
+                    }else {
```

Comment:
Formatting. Space is missing before the else.
Suggestion: Use the IDE to format the code. Like this: Code > Reformat Code